workspaces/src/rpc/client.rs

```diff
@@ -373,22 +373,22 @@ async fn cached_nonce(nonce: &AtomicU64, client: &Client) -> Result<(CryptoHash,
 /// into contention with others.
 async fn fetch_tx_nonce(
     client: &Client,
-    account_id: AccountId,
-    public_key: near_crypto::PublicKey,
+    cache_key: &(AccountId, near_crypto::PublicKey),
```

Comment:
same with this, feels like you avoid a clone if you just have this be an owned value and not take a ref outside?

Reply:
Same with the other one with trying to optimize a little bit. But my thought was that since `fetch_tx_nonce` only really needs to clone on writing the nonce with key initially, it'd be better to not give it owned values everytime. Especially more so when `send_batch_tx_and_retry` can take the actual owned value later.

The most recent commit I had should make better due of this situation though. Moved the initial creation of the cache_key outside of the retry loop, so we can just always take a reference to it, and removed the need to take ownership of the cache_key when merely just deleting the entry from the nonce map.